CRM/Dedupe/Merger.php

```diff
@@ -1631,41 +1632,57 @@ public static function getRowsElementsAndInfo(int $mainID, int $otherID, bool $c
     }
 
     // handle custom fields
-    $mainTree = self::getTree($main['contact_type'], $mainID,
-      $main['contact_sub_type'] ?? NULL,
-      $checkPermissions ? CRM_Core_Permission::EDIT : FALSE
-    );
-    $otherTree = self::getTree($main['contact_type'], $otherID,
-      $other['contact_sub_type'] ?? NULL,
-      $checkPermissions ? CRM_Core_Permission::EDIT : FALSE
-    );
-
-    foreach ($otherTree as $gid => $group) {
-      if (!isset($group['fields'])) {
-        continue;
+    $filters = [
+      'extends' => $main['contact_type'],
+      'is_active' => TRUE,
+      'extends_entity_column_value' => NULL,
```

Comment:
This also works if you add to the filters (per comments below)
![image](https://github.com/civicrm/civicrm-core/assets/336308/8f68d33e-0e12-48ae-ac36-8de24513fb34)